custom_components/amplipi/media_player.py

```diff
@@ -56,6 +58,31 @@
 
 PARALLEL_UPDATES = 1
 
+class AmpliPiCoordinator(DataUpdateCoordinator):
+    def __init__(self, hass, config_entry, my_api):
+        super().__init__(
+            hass,
+            _LOGGER,
+            config_entry=config_entry,
+            name="hacs_amplipi",
+            update_interval=timedelta(seconds=2),
+            always_update=True
+        )
+        self.api = my_api
+
+    async def _async_update_data(self):
+        """Fetch data from API endpoint.
+
+        This is the place to pre-process the data to lookup tables
+        so entities can quickly look up their data.
+        """
+        try:
+            state = await self.api.get_status()
+            return state
+        except Exception as e:
+            raise UpdateFailed(f"Error fetching data: {e}")
+
+
```

Comment:
Nice! Looking at the HA docs they suggest adding it to a separate file called coordinator.py

Data update coordinator - coordinator.py
There are multiple ways for your integration to receive data, including push or poll. Commonly integrations will fetch data with a single coordinated poll across all entities, which requires the use of a DataUpdateCoordinator. If you want to use one, and you choose to create a subclass of it, it is recommended to define the coordinator class in coordinator.py. [More information about DataUpdateCoordinator](https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities).